test-case/stress-reboot.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+##
+## Case Name: stress test reboot
+## Preconditions:
+##    N/A
+## Description:
+##    run reboot for the stress test
+##    default duration is 10s
+##    default loop count is 3
+## Case step:
+##    1. check system status is correct
+##    2. wait for the Random value
+##    3. trigger for the reboot
+## Expect result:
+##    Test execute without report error in the LOG
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['l']='loop'     OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1         OPT_VALUE_lst['l']=5
+
+func_opt_parse_option $*
+
+func_lib_check_sudo
+func_lib_trigger_stress
+
+loop_count=${OPT_VALUE_lst['l']}
+
+if [ $loop_count -le 0 ]; then
+    mv $LOG_ROOT/cmd-orig.txt $LOG_ROOT/cmd.txt -f
+    exit 0
+fi
+# write the orig & status count to the status file
+echo $loop_count >> $STRESS_STATUS_LOG
+current=$(wc -l $STRESS_STATUS_LOG|awk '{print $1;}')
+count=$(head -n 1 $STRESS_STATUS_LOG|awk '{print $1;}')
+# because write record by desc
+current=$[ $count - $current + 1 ]
+
+dlogi "Round: $current/$count" | tee -a $STRESS_OUTPUT_LOG
+
+declare -a verify_lst
+verify_lst=(${verify_lst[*]} "verify-firmware-presence.sh")
+verify_lst=(${verify_lst[*]} "verify-kernel-module-load-probe.sh")
+verify_lst=(${verify_lst[*]} "verify-pcm-list.sh")
+verify_lst=(${verify_lst[*]} "verify-sof-firmware-load.sh")
+verify_lst=(${verify_lst[*]} "verify-tplg-binary.sh")
+
+# here will write 2 line into $STRESS_OUTPUT_LOG
+# because for the 1st round those information write by nohup command redirect
+# but after system reboot the nohup redirect is missing
+dlogi "Check status begin" | tee -a $STRESS_OUTPUT_LOG
+for i in ${verify_lst[*]}
+do
+    #$(dirname ${BASH_SOURCE[0]})/$i
+    if [ 0 -ne 0 ];then
+        # last line add failed keyword
+        sed -i '$s/$/ fail/' $STRESS_STATUS_LOG
+        dlogi "$i: fail" |tee -a $STRESS_OUTPUT_LOG
+        exit 1
+    else
+        # last line add pass keyword
+        dlogi "$i: pass" |tee -a $STRESS_OUTPUT_LOG
+    fi
+done
+dlogi "Status check finished" | tee -a $STRESS_OUTPUT_LOG
+sed -i '$s/$/ pass/' $STRESS_STATUS_LOG
+
+# run the script for the next round
+loop_count=$[ $loop_count - 1 ]
+
+# Store the environment for the next round load this script
+dlogi "Setup the environment for next round bootup" |tee -a $STRESS_OUTPUT_LOG
+sudo sof-boot-once.sh export LOG_ROOT=$LOG_ROOT
+# just keep the last dmesg bootup kernel line for the success
+sudo sof-boot-once.sh export DMESG_LOG_START_LINE=$(wc -l /var/log/kern.log|awk '{print $1;}')
```

Comment:
Why call `sof-boot-once.s` multiple times? Maybe some re-name for this helper function is needed.

Reply:
convert sof-boot-once.sh to the help function?  
but I think it maybe useful for the stand along, expect we just handle it for the reboot

here will cause other problem: how to clean up the rc.local file  
Now sof-boot-once.sh will do the write and cleanup rc.local it also is reason why this script is stand alone